src/sensai/featuregen.py

```diff
@@ -86,6 +103,11 @@ def generate(self, df: pd.DataFrame, ctx=None) -> pd.DataFrame:
                 if series.dtype.name != 'category':
                     resultDF[colName] = series.astype('category', copy=False)
 
+        self.generatedColumnNames = resultDF.columns
+        if self.generatedColumnsRuleTemplate is not None:
+            self._generatedColumnsRule = data_transformation.DFTNormalisation.Rule(r"(%s)" % "|".join(self.generatedColumnNames), **self.generatedColumnsRuleTemplate.__dict__)
+            self._normalisationRules.append(self._generatedColumnsRule)
```

Comment:
Is anything preventing this from being executed more than once?

Reply:
Whoops, that was a rather big mistake. I have fixed it by separating rules for categorical features and for the rest